modules/smilewantedBidAdapter.js

```diff
@@ -0,0 +1,84 @@
+import * as utils from '../src/utils';
+import { config } from '../src/config';
+import { registerBidder } from '../src/adapters/bidderFactory';
+
+export const spec = {
+  code: 'smilewanted',
+  aliases: ['smile', 'sw'],
+
+  isBidRequestValid: function(bid) {
+    return !!(bid.params && bid.params.zoneId);
+  },
+
+  buildRequests: function(validBidRequests, bidderRequest) {
+    return validBidRequests.map(bid => {
+      var payload = {
+        zoneId: bid.params.zoneId,
+        currencyCode: config.getConfig('currency.adServerCurrency'),
+        bidfloor: bid.params.bidfloor || 0.0,
+        tagId: bid.adUnitCode,
+        sizes: bid.sizes.map(size => ({
+          w: size[0],
+          h: size[1]
+        })),
+        pageDomain: utils.getTopWindowUrl(),
```

Comment:
utils.getTopWindowUrl() is going to be depreciated. There is a property 'bidderRequest.refererInfo' which can provide the referral info. Can you use that instead of this?
Please refer to : http://prebid.org/dev-docs/bidder-adaptor.html#referrers

Reply:
We updated the referrer recovery method and modified unit tests to prevent the case where there is no referrer, which is the case when we did the "gulp test" command, it returned the error `Cannot read property" refererInfo' of undefined`, I don't know if it's intended ?

Reply:
@MaxSmileWanted Would you be able to point out which unit test failed and (depending on the size) provide a copy of the code in they way you attempted to modify it?  

The `refererInfo` exists on the `bidderRequest` object, which is the 2nd argument for the `spec.buildRequests()` function.  

So without seeing the changes, it's likely that either:
- the unit test's `spec.buildRequests()` call didn't populate the `bidderRequest` argument
- the supplied `bidderRequest` object was just missing its own `refererInfo` property and needed to setup with some mocked values

Reply:
@jsnellbaker Yes you right the bidderRequest argument it's not populate through the unit test, so i append a forced value with this code in an another "it" :
```JAVASCRIPT
const request = spec.buildRequests(DEFAULT_PARAMS, {
  refererInfo: {
    referer: 'http://localhost/Prebid.js/integrationExamples/gpt/hello_world.html'
  }
});
```